### ISSUE TITLE
Manage salt configuration from gitfs

### ISSUE THREAD

**Issue by tobiasBora** [saltstack/salt#45043]
### Description of Issue/Question

Hello,

I'd like to use one git repository that would contain all (or most of) the code that is needed for it to run. In an ideal world, I'd like to be able to setup once the configuration in /etc/salt/master and then only use the git repo.

But for now it's not possible, because I can't find any way to synchronize the master configuration with a git repo, and it means that I need to manually do crazy hack, like adding a hook to my git repo that rsync the folder  /etc/salt/master.d. And why do I need to change this directory when I build my project ? Just because all the rootfs/gitfs stuff is set inside, and that my formulas change over the time.

Did I miss something, or rsync is indded the best way to proceed for now ?

Thank you!

**Comment by gtmanfred** [comment]
This is a duplicate of #11724

We would love to see a PR for this if you had time.

Thanks,
Daniel

**Comment by tobiasBora** [comment]
Thank you for your answer. I'd love to write a PR, but I don't know at all how salt works inside, and for now I can't spend 50h on that. I tried to give it a quick look but I think that I can't do it with just 10 lines, because it needs to run before everything, including the server loading configuration.

Meanwhile, I wrote my own script that bootstrap a node, and do all the dirty job [here](https://github.com/tobiasBora/all_saltgit), if someone else needs it.
